libbeat/publisher/processing/default.go

```diff
@@ -356,7 +357,7 @@ func (b *builder) Create(cfg beat.ProcessingConfig, drop bool) (beat.Processor,
 	}
 
 	// setup 10: debug print final event (P)
-	if b.log.IsDebug() {
+	if b.log.IsDebug() || publisher.UnderAgent() {
```

Comment:
Why is the `publisher.UnderAgent()` check necessary here?

Reply:
I'm also wondering the same, the "Publish event" logs also flood logs of standalone Beats, often rendering them unusable. I believe we should only use a trace level, regardless if the Beat is running under Agent or not.

Reply:
It is needed because when running under Elastic Agent that processor needs to always be added, so it can be dynamically toggled when the log level changes without needed to reload the entire configuration.